webpack/assets/javascripts/react_app/components/PasswordStrength/__tests__/PasswordStrength.test.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import toJson from 'enzyme-to-json';
+
+import PasswordStrength from '../PasswordStrength';
+
+describe('PasswordStrength component', () => {
+  const getBasePrpops = () => ({
+    updatePassword: jest.fn(),
+    updatePasswordConfirmation: jest.fn(),
+    doesPasswordsMatch: true,
+    data: {
+      className: 'some-class-name',
+      id: 'some-id',
+      name: 'some-name',
+      error: 'some-password-error',
+      userInputIds: [],
+    },
+  });
+
+  beforeEach(() => {
+    document.__originalGetElementById = document.getElementById;
+    document.getElementById = jest.fn(id => ({ value: id }));
+  });
+
+  afterEach(() => {
+    document.getElementById = document.__originalGetElementById;
+    delete document.__originalGetElementById;
+  });
+
+  describe('rendering', () => {
+    test('it should get rendered correctly', () => {
+      const component = shallow(<PasswordStrength {...getBasePrpops()} />);
+
+      expect(toJson(component)).toMatchSnapshot();
+    });
+
+    test('it should get rendered correctly with password-confirmation', () => {
+      const props = getBasePrpops();
+      props.data.verify = {
+        name: 'user[password_confirmation]',
+        error: 'some-password-confirmation-error',
+      };
+
+      const component = shallow(<PasswordStrength {...props} />);
+
+      expect(toJson(component)).toMatchSnapshot();
+    });
+
+    test('it should get rendered correctly with unmatched password-confirmation', () => {
+      const props = { ...getBasePrpops(), doesPasswordsMatch: false };
+      props.data.verify = {
+        name: 'user[password_confirmation]',
+        error: 'some-password-confirmation-error',
+      };
+
+      const component = shallow(<PasswordStrength {...props} />);
+
+      expect(toJson(component)).toMatchSnapshot();
+    });
+
+    test('it should get rendered correctly with user-input-ids', () => {
+      const props = getBasePrpops();
+      props.data.userInputIds = ['input1', 'input2'];
+
+      const component = shallow(<PasswordStrength {...props} />);
+
+      expect(toJson(component)).toMatchSnapshot();
+      expect(document.getElementById.mock.calls).toMatchSnapshot('getElementById calls');
```

Comment:
doesn't it better to use `toBeCalled` function in this case? (instead of snapshot)

Reply:
Then I will only know that the function was called.

The snapshot contains the fact that the functions called twice and the arguments from each call:
https://github.com/theforeman/foreman/pull/5240/files#diff-04cd101c699dfc26787bd687fc50fba2R194